lib/aliquot-pay/util.rb

```diff
@@ -11,18 +11,18 @@ def self.generate_shared_secret(private_key, public_key)
       private_key.dh_compute_key(public_key)
     end
 
-    def self.derive_keys(ephemeral_public_key, shared_secret, info)
+    def self.derive_keys(ephemeral_public_key, shared_secret, info, length: 32)
       input_keying_material = ephemeral_public_key + shared_secret
       if OpenSSL.const_defined?(:KDF) && OpenSSL::KDF.respond_to?(:hkdf)
         h = OpenSSL::Digest::SHA256.new
-        hbytes = OpenSSL::KDF.hkdf(input_keying_material, hash: h, salt: '', length: 32, info: info)
+        hbytes = OpenSSL::KDF.hkdf(input_keying_material, hash: h, salt: '', length: length * 2, info: info)
```

Comment:
So 64 is correct and 32 was incorrect? Did it work before?

Reply:
Both 64 and 32 are correct.
For `ECv1` you need to generate 32 bytes of key material, 128 bits for AES key and 128 bits for MAC key.
For `ECv2`you need 64 bytes, 256 bits for each.

Reply:
Since `ECv2` has superseded `ECv1` the required 64 bytes of keys is now default.

Reply:
With v1 I suspect?

Reply:
What do you mean ct?
Yes, it worked before with `ECv1`.